server/controllers/artGalleryController.js

```diff
@@ -0,0 +1,39 @@
+const AWS = require('aws-sdk');
+const Submission = require('../models/submissionModel');
+
+// Connect to the AWS S3 Storage
+const s3 = new AWS.S3({
+  accessKeyId: process.env.ACCESS_KEY_ID,
+  secretAccessKey: process.env.SECRET_ACCESS_KEY,
+  region: process.env.S3_REGION,
+});
+
+const getAllSubmissions = async (req, res) => {
+  try {
+    // S3 Key retrieval from MongoDB
+    // Empty `filter` means "match all documents"
+    const filter = {};
+    const allSubmissions = await Submission.find(filter);
+
+    // Image retrieval from AWS S3
+    const s3Promises = await allSubmissions.map(async (submission) => s3.getObject({
+      Bucket: process.env.S3_BUCKET,
+      Key: submission.s3keys[0],
+    }).promise());
+    const imageDataList = Promise.all(s3Promises);
```

Comment:
Not sure if you could just add `await` before `s3.getObject` and remove the promise aspect. If you can, then you should be able to get all the imageDataList, otherwise, looks fine